### ISSUE TITLE
feat: @koi/ipc-nexus — agent-to-agent messaging

### ISSUE THREAD

**Issue by windoliver** [windoliver/koi#192]
## Summary

Integrate Koi agents with Nexus's filesystem-based IPC system. Agents communicate via Nexus inboxes (JSON files in `/agents/{id}/inbox/`) with hot path (NATS) + cold path (filesystem) delivery.

**This is the critical missing piece** — Koi's architecture says "no direct agent-to-agent communication, agents interact through infrastructure." Nexus IS that infrastructure, but there's no integration surface.

### 1. Add `AgentMailbox` singleton component

```typescript
// ecs.ts — new singleton component
interface AgentMailbox {
  readonly send: (to: string, payload: unknown, type?: string) => Promise<Result<void, KoiError>>;
  readonly onMessage: (handler: (message: AgentMessage) => Promise<void>) => () => void;
}

interface AgentMessage {
  readonly id: string;
  readonly from: string;
  readonly to: string;
  readonly type: string;
  readonly payload: unknown;
  readonly timestamp: number;
}

const MAILBOX: SubsystemToken<AgentMailbox> = token<AgentMailbox>("mailbox");
```

**Design rationale — why this is slim enough for L0:**
- `send()` + `onMessage()` — same pattern as `ChannelAdapter` (send + onMessage)
- No inbox/outbox/dead_letter concepts — those are Nexus implementation details
- No TTL, no signing, no delivery mode — all L2 concerns
- No envelope format — `AgentMessage` is the minimal shared vocabulary

### 2. Make `ComponentProvider.attach` async + add `detach`

```typescript
interface ComponentProvider {
  readonly name: string;
  readonly attach: (agent: Agent) => ReadonlyMap<string, unknown> | Promise<ReadonlyMap<string, unknown>>;
  readonly detach?: (agent: Agent) => void | Promise<void>;
}
```

**Why:** Nexus IPC provider needs to provision `/agents/{id}/inbox/` during attach (async I/O) and drain/cleanup during detach. Current L0 is sync-only, blocking all remote backends.

### What it implements

| L0 Interface | Nexus Backend |
|---|---|
| `AgentMailbox.send()` | Nexus `MessageSender` — writes to `/agents/{to}/inbox/{ts}_{id}.json` |
| `AgentMailbox.onMessage()` | Nexus `MessageProcessor` — polls inbox or subscribes via EventBus/NATS |
| `ComponentProvider.attach()` | Calls `AgentProvisioner.provision()` — creates inbox/outbox/processed/dead_letter dirs |
| `ComponentProvider.detach()` | Updates agent status, drains pending messages |

### Nexus features consumed (L2 only — NOT in L0)

- `MessageEnvelope` format with correlation IDs, TTL, Ed25519 signing
- Hot path (NATS) + cold path (filesystem) delivery modes
- TTL sweeper for expired messages → dead_letter
- Deduplication via CacheStoreABC
- Backpressure (inbox size limit: 1000 messages)
- Zone-scoped isolation

### Configuration

```yaml
# koi.yaml
ipc:
  backend: "@koi/ipc-nexus"
  options:
    nexusUrl: "http://localhost:8000"
    zone: "default"
    deliveryMode: "hot_cold"  # or "cold_only", "hot_only"
    inboxLimit: 1000
    signing: true
```

### Alternative L2 backends (same L0 interface)

- `@koi/ipc-memory` — in-process array + callback (tests)
- `@koi/ipc-redis` — Redis pub/sub (lightweight alternative)

## Acceptance Criteria

- [ ] `AgentMailbox` and `AgentMessage` types added to `@koi/core`
- [ ] `MAILBOX` well-known token exported
- [ ] `ComponentProvider` made async-compatible with optional `detach`
- [ ] `@koi/ipc-nexus` package created with NexusIPCProvider
- [ ] Agent A can send a message to Agent B via `agent.component(MAILBOX).send()`
- [ ] Messages persist across restarts (cold path)
- [ ] Hot path delivery via NATS when available
- [ ] In-memory test backend for unit tests
- [ ] 80%+ test coverage

## References

- Nexus IPC: `~/nexus/src/nexus/ipc/delivery.py` (MessageSender, MessageProcessor)
- Nexus provisioning: `~/nexus/src/nexus/ipc/provisioning.py`
- Nexus envelope: `~/nexus/src/nexus/ipc/envelope.py`
- Koi architecture: `docs/architecture/Koi.md` lines 523-533 (agent-to-agent communication)
